### PR TITLE
In Soviet Nanotransen, Tram consume you!

### PR DESCRIPTION
**Pull request by DragonTrance**
## About The Pull Request
Stops the tram hilariously qdel-ing the SM, and now makes the SM consume the tram instead.

Adds a blacklist typepath var of things that all lifts/trams cannot interact with, which lift master datums use.
Also makes it so the tram control console cannot be consumed by the SM

## Why It's Good For The Game
Bug fix, but also this could be funny seeing someone drag the SM shard around

Turns this boring, sad thing...

Into this fun, exciting thing!
https://user-images.githubusercontent.com/26515331/147429544-ea0722bd-4a19-4404-bba9-3abb4fe5ad3f.mp4

## Changelog

:cl:
fix: The tram can no longer consume and delete the Supermatter.
/:cl:

### CODE REVIEW

File: code/game/machinery/computer/tram_controls.dm [hunk 1]
```diff
@@ -4,7 +4,7 @@
 	icon_screen = "tram"
 	icon_keyboard = "atmos_key"
 	circuit = /obj/item/circuitboard/computer/tram_controls
-	flags_1 = NODECONSTRUCT_1
+	flags_1 = NODECONSTRUCT_1 | SUPERMATTER_IGNORES_1 //What fun is it if we get consumed by the SM and can't move the tram anymore?
```
Comment (JohnFulpWillard) [inline code/game/machinery/computer/tram_controls.dm:7]: kinda dissapointed in this

**Comment by SuperNovaa41** [comment]
this looks funny

File: code/game/machinery/computer/tram_controls.dm (hunk 1 above)
Comment (DragonTrance) [reply code/game/machinery/computer/tram_controls.dm:7]: I mean, it's already indestructible by other flags. This is only to make it so it doesn't dust when the supermatter rams into it

**Review by MrMelbert — CHANGES_REQUESTED** [review]
Minor stuff

**Pull request merged**